File: src/main/java/com/example/MediTime/dto/ClientMedicationDTO.java
// Amy Wickham 121785021
// File: ClientMedicationDTO.java
// Description: DTO for assigning and retrieving client-medication data.

package com.example.meditime.dto;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

public class ClientMedicationDTO {

    // Getters and Setters
    @Setter
    @Getter
    private Long clientId;

    @Setter
    @Getter
    private Long medicationId;

    @Getter
    @Setter
    private String dosage;
    @Setter
    @Getter
    private LocalDate startDate;
    @Getter
    @Setter
    private LocalDate endDate;
    private boolean isPaused = false;
    @Setter
    @Getter
    private List<LocalTime> scheduledTimes;

    private List<ClientMedicationDTO> medications;

    public List<ClientMedicationDTO> getMedications() {
        return medications;
    }

    public void setMedications(List<ClientMedicationDTO> medications) {
        this.medications = medications;
    }

    public boolean isPaused() {
        return isPaused;
    }

    public void setPaused(boolean paused) {
        isPaused = paused;
    }

}

File: Mobile/app/src/main/java/com/example/medicationapp/model/dto/ClientMedicationDTO.kt
package com.example.medicationapp.model.dto

import java.io.Serializable
import java.time.LocalDate
import java.time.LocalTime

/**
 * Data Transfer Object (DTO) representing a client's medication information.
 *
 * Used to transfer combined medication and client details, typically between layers of the app
 * (e.g., from backend to UI or between components).
 *
 * Fields:
 * - clientId: ID of the client prescribed the medication.
 * - medicationId: ID of the medication.
 * - dosage: Dosage instructions (e.g., "2 tablets daily").
 * - clientName: Full name of the client.
 * - medicationName: Name of the medication.
 * - startDate: Date when the medication regimen begins.
 * - endDate: Date when the medication regimen ends.
 * - isPaused: Indicates if the medication is currently paused.
 * - scheduledTimes: List of times the medication should be taken daily.
 *
 * Implements Serializable for easy data passing (e.g., through Intents or Bundles).
 */

data class ClientMedicationDTO(
    val clientId: Long,
    val medicationId: Long,
    val dosage: String,
    val clientName: String,
    val medicationName: String,
    val startDate: LocalDate,
    val endDate: LocalDate,
    val isPaused: Boolean,
    val scheduledTimes: List<LocalTime>
): Serializable